core/io/resource.h

```diff
@@ -90,6 +90,8 @@ class Resource : public Reference {
 	static Node *(*_get_local_scene_func)(); //used by editor
 
 	virtual bool editor_can_reload_from_file();
+	virtual void reset_state(); //for resources that use variable amount of properties, either via _validate_property or _get_property_list, this function needs to be implemented to correctly clear state
```

Comment:
Just a tip for future comments, when it's so long it would be nicer to have it before the method than as trailing comment:
```suggestion
	 // For resources that use variable amount of properties, either via _validate_property or _get_property_list,
	 // this function needs to be implemented to correctly clear state.
	virtual void reset_state();
```